AutofacContrib.NSubstitute/SubstituteForBuilder.cs

```diff
@@ -28,6 +29,43 @@ public AutoSubstituteBuilder Configure(Action<TService> action)
             action(_service);
             return _builder;
         }
-    }
 
+        /// <summary>
+        /// Allows for configuration of the service with access to the resolved components.
+        /// </summary>
+        /// <param name="action">The delegate to configure the service.</param>
+        /// <returns>The original <see cref="AutoSubstituteBuilder"/>.</returns>
+        public AutoSubstituteBuilder Configure(Action<TService, IComponentContext> action)
+        {
+            _builder.ConfigureBuilder(builder =>
+            {
+                builder.RegisterBuildCallback(scope =>
+                {
+                    action(_service, scope);
+                });
+            });
+
+            return _builder;
+        }
+
+        /// <summary>
+        /// Completes the configuration of the substitute.
+        /// </summary>
+        /// <returns>The original <see cref="AutoSubstituteBuilder"/>.</returns>
+        public AutoSubstituteBuilder Configured()
+        {
+            return _builder;
+        }
+
+        /// <summary>
+        /// Allows for a similar pattern as other Provide methods.
+        /// </summary>
+        /// <param name="service">Parameter to obtain a provided value.</param>
+        /// <returns></returns>
+        public SubstituteForBuilder<TService> Provide(out TService service)
```

Comment:
I'm a bit confused about this. The other `Provide` methods all provide something to the container e.g. a type or an instance, but this one simply has an `out` parameter. Am I missing something?

Reply:
Provide may not be the best method name. I'm looking to have a way to capture the value that was configured without having to resolve anything else. Open to suggestions

Reply:
I've switched it to `.GetValue(...)`. Again, happy for suggestions